File: q27.py
#code
from collections import defaultdict
t = int(input())
for _ in range(t):
    m,n = list(map(int,input().strip().split()))
    arr = list(map(int,input().strip().split()))
    sub = list(map(int,input().split()))
    a = defaultdict(int)
    s = defaultdict(int)
    for i in arr:
        a[i] += 1
    for j in sub:
        s[j] += 1
    flag = 0
    for k,v in s.items():
        if(v!=a[k]):
            print("No")
            flag = 1
            break
    if(flag == 0):
        print("Yes")

File: q27.cpp
// Array Subset of another array
// This code passes one test case in vscode but does not pass that case in gfg
// https://www.geeksforgeeks.org/find-whether-an-array-is-subset-of-another-array-set-1/
// very interesting solutions mentioned in gfg
# include<bits/stdc++.h>
#include<map>
using namespace std;

int main()
{
    int t,i,j,m,n,a,flag = 0;
    map<int, int> arr, sub;
    cin>>t;
    for(i=0;i<t;i++)
    {
        cin>>m>>n;
        for(j=0;j<m;j++)
        {
            cin>>a;
            arr[a] += 1;
        }
        for(j=0;j<n;j++)
        {
            cin>>a;
            sub[a] += 1;
        }
        for(auto i:arr)
        {
            cout<<i.first<<i.second<<"\n";
        }
        for(auto i:sub)
        {
            cout<<i.first<<i.second<<"\n";
        }
        for(auto i:sub)
        {
            cout<<i.first<<i.second<<arr[i.first]<<"\n";
            if(i.second != arr[i.first])
            {
                flag = 1;
                printf("No\n");
                break;
            }
        }
        if(flag==0)
            printf("Yes\n");
        arr.clear();
        sub.clear();
    }
}